src/test-project/src/test/java/com/redhat/devtools/intellij/commonUiTestLibrary/fixturesTest/dialogs/projectManipulation/NewProjectDialogTest.java

```diff
@@ -11,145 +11,284 @@
+    private NewProjectDialogFirstPage newProjectDialogFirstPage;
+    private MainIdeWindow mainIdeWindow;
 
     @BeforeEach
     public void openNewProjectDialog() {
         openNewProjectDialogFromWelcomeDialog();
-        newProjectDialog = remoteRobot.find(NewProjectDialog.class, Duration.ofSeconds(10));
+        newProjectDialogWizard = remoteRobot.find(NewProjectDialogWizard.class, Duration.ofSeconds(10));
+        newProjectDialogFirstPage = newProjectDialogWizard.find(NewProjectDialogFirstPage.class, Duration.ofSeconds(10));
     }
 
     @AfterEach
     public void cleanUp() {
-        try {
-            // tests ending with opened New Project Dialog needs to close the dialog
-            remoteRobot.find(NewProjectDialog.class, Duration.ofSeconds(10)).cancel();
-        } catch (WaitForConditionTimeoutException e) {
+        if (mainIdeWindow != null) {
+            // tests ending with opened Main Ide Window needs to close the project and clear workspace
+            IdeStatusBar ideStatusBar = mainIdeWindow.find(IdeStatusBar.class, Duration.ofSeconds(10));
+            ideStatusBar.waitUntilProjectImportIsComplete();
+            TipDialog.closeTipDialogIfItAppears(remoteRobot);
+            mainIdeWindow.maximizeIdeWindow();
+            ideStatusBar.waitUntilAllBgTasksFinish();
+            mainIdeWindow.closeProject();
+            remoteRobot.find(FlatWelcomeFrame.class, Duration.ofSeconds(10)).clearWorkspace();
+            mainIdeWindow = null;
+        } else {
             try {
-                // tests ending with opened Main Ide Window needs to close the project and clear workspace
-                remoteRobot.find(MainIdeWindow.class, Duration.ofSeconds(5));
-                closeProject();
-                FlatWelcomeFrame flatWelcomeFrame = remoteRobot.find(FlatWelcomeFrame.class, Duration.ofSeconds(10));
-                flatWelcomeFrame.clearWorkspace();
-            } catch (WaitForConditionTimeoutException e2) {
+                // tests ending with opened New Project Dialog needs to close the dialog
+                newProjectDialogWizard.cancel();
+            } catch (WaitForConditionTimeoutException e) {
                 // tests ending with opened Flat Welcome Frame does not need any assistance
             }
         }
     }
 
     @Test
     public void setProjectNamePlainJavaProjectTest() {
-        testProjectNameInputField(plainJavaProjectName, NewProjectType.PLAIN_JAVA);
+        testProjectNameInputField(NewProjectType.PLAIN_JAVA);
     }
 
     @Test
     public void setProjectNameMavenProjectTest() {
-        testProjectNameInputField(mavenProjectName, NewProjectType.MAVEN);
+        testProjectNameInputField(NewProjectType.MAVEN);
     }
 
     @Test
     public void setProjectNameGradleProjectTest() {
-        testProjectNameInputField(gradleProjectName, NewProjectType.GRADLE);
+        testProjectNameInputField(NewProjectType.GRADLE);
+    }
+
+    @Test
+    public void javaMoreSettingsTest() {
```

Comment:
why is this test method so long? couldn't it be divided into multiple test methods with appropriate (could be long) names?

Reply:
it also counts for other test methods. I would like to see that if something fails in the test, I can read it from the test method ideally. Of course, it does not mean that it is a rule for every case.